lib/auth/auth_with_roles.go

```diff
@@ -36,33 +36,33 @@ import (
 	"github.com/tstranex/u2f"
 )
 
-// AuthWithRoles is a wrapper around auth service
+// WithRoles is a wrapper around auth service
 // methods that focuses on authorizing every request
-type AuthWithRoles struct {
-	authServer *AuthServer
+type WithRoles struct {
```

Comment:
Nit: `WithRoles` sounds like a function name to me.

Reply:
Renamed to `ServerWithRoles` :+1: